File: Data Structures & Algorithms/majority-element/submission-4.java
class Solution {
    public int majorityElement(int[] nums) {
        int candidate = 0;
        int majority = 0;

        for(int num : nums){
            if(num == candidate){
                majority += 1;
            }
            else if(majority == 0){
                candidate = num;
                majority = 1;
            }
            else{
                majority -= 1;
            }
        }
        return candidate;
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-4.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        current_sum = 0
        freqmap = {0:1}

        for num in nums:
            current_sum += num

            previous_current_sum = current_sum - k

            count += freqmap.get(previous_current_sum, 0)

            freqmap[current_sum] = freqmap.get(current_sum, 0) + 1
        return count